libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx

```diff
@@ -49,22 +49,32 @@ export const RecommendationPanelContext = (props: RecommendationPanelContextProp
 
   const [selectionState, setSelectionState] = useState<SelectionState>(SELECTION_STATES.SEARCH);
 
-  const selectedOperationId = useSelectedSearchOperationId();
   const { data: allOperations, isLoading: isLoadingOperations } = useAllOperations();
-  const selectedOperation = allOperations.find((o) => o.id === selectedOperationId);
+  const [selectedOperation, setSelectedOperation] = useState<DiscoveryOperation<DiscoveryResultTypes> | undefined>(undefined);
 
   const selectedOperationGroupId = useSelectedSearchOperationGroupId();
   const { data: allConnectors } = useAllConnectors();
   const selectedConnector = allConnectors?.find((c) => c.id === selectedOperationGroupId);
 
+  // effect to set the current list of operations by group
   useEffect(() => {
-    if (!allOperations || !selectedOperationGroupId) return;
-    const filteredOps = allOperations.filter((operation) => {
-      const apiId = operation.properties.api.id;
-      return areApiIdsEqual(apiId, selectedOperationGroupId);
+    if (!selectedOperationGroupId) return;
+
+    const searchOperation = SearchService().getOperationsByConnector?.bind(SearchService());
+
+    const searchResultPromise = searchOperation
+      ? searchOperation(selectedOperationGroupId, filters['actionType']?.toLowerCase())
+      : Promise.resolve(
+          (allOperations ?? []).filter((operation) => {
+            const apiId = operation.properties.api.id;
+            return areApiIdsEqual(apiId, selectedOperationGroupId);
+          })
+        );
+
+    searchResultPromise.then((filteredOps) => {
+      setAllOperationsForGroup(filteredOps);
+      setSelectionState(SELECTION_STATES.DETAILS);
     });
-    setAllOperationsForGroup(filteredOps);
-    setSelectionState(SELECTION_STATES.DETAILS);
   }, [selectedOperationGroupId, allOperations]);
```

Comment:
Would be good to address the warning `React Hook useEffect has a missing dependency: 'filters'. Either include it or remove the dependency array`

It might be worth splitting this up into separate useEffects for if `searchOperation` is defined vs. not defined? My concern is that if anything in the dep array changes, for us this would cause a call to the host (and possibly interfere with the saved results list in PPUX)

Reply:
I don't believe the split is necessary, the filters dropdown lists don't show up on the connector view.